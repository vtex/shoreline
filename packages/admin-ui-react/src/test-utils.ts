import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
