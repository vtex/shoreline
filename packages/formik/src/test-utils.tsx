import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
