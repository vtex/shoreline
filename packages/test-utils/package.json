{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
