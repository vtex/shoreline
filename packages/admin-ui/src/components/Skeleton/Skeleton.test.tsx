import React from 'react'
import { render, axe } from '../../test-utils'

import { Skeleton } from './index'

describe('Skeleton tests', () => {
  it('should not have a11y violations', async () => {
    const { container } = render(<Skeleton />)

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
