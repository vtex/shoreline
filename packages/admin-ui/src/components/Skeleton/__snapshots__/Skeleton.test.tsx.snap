// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton tests should match snapshot 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    -webkit-background-position: -200px 0;
    background-position: -200px 0;
  }

  100% {
    -webkit-background-position: calc(200px + 100%) 0;
    background-position: calc(200px + 100%) 0;
  }
}

.emotion-0 {
  display: inline-block;
  width: 100%;
  height: 100%;
  background: linear-gradient(90deg, hsla(0, 0%, 90%, 1), hsla(0, 0%, 87%, 1)),hsla(0, 0%, 90%, 1);
  -webkit-background-size: 200px 100%;
  background-size: 200px 100%;
  background-repeat: no-repeat;
  line-height: 1;
  border-radius: 4px;
  -webkit-animation: animation-0 1.2s ease-in-out infinite;
  animation: animation-0 1.2s ease-in-out infinite;
}

<div
    class="emotion-0"
  />
  @keyframes animation-0 {
  0% {
    -webkit-background-position: -200px 0;
    background-position: -200px 0;
  }

  100% {
    -webkit-background-position: calc(200px + 100%) 0;
    background-position: calc(200px + 100%) 0;
  }
}

.emotion-0 {
  display: inline-block;
  width: 100%;
  height: 100%;
  background: linear-gradient(90deg, hsla(0, 0%, 90%, 1), hsla(0, 0%, 87%, 1)),hsla(0, 0%, 90%, 1);
  -webkit-background-size: 200px 100%;
  background-size: 200px 100%;
  background-repeat: no-repeat;
  line-height: 1;
  border-radius: 100%;
  -webkit-animation: animation-0 1.2s ease-in-out infinite;
  animation: animation-0 1.2s ease-in-out infinite;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;
